src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -27,11 +28,16 @@ internal class FunctionExecutor : IFunctionExecutor
         private readonly IWebJobsExceptionHandler _exceptionHandler;
         private readonly TraceWriter _trace;
         private readonly IAsyncCollector<FunctionInstanceLogEntry> _fastLogger;
+        private readonly ILogger _logger;
```

Comment:
Yep, I was thinking we could have the executor emit events (started, completed, completedWithError) and each logger could subscribe to them in order to move all this out of the executor. Killing off tracewriter would be a big help.